### DISCUSSION TITLE
Respond with UNAUTHORIZED for APIs

### DISCUSSION THREAD

**Discussion by k-bx** [maxcountryman/axum-login#238] [General]
I'm using axum-login for both, HTML endpoints and API endpoints. While `307 Temporary Redirect` into `/login?next=%2Fapi%2Fuser` makes sense for HTML endpoints, for JSON API endpoints it'd be better to return `401 UNAUTHORIZED` response with details in body.

What's the best way to do this atm?

**Comment by maxcountryman** [comment]
For this I would recommend writing your own version of `login_required` (it's wrapping [`from_fn`](https://docs.rs/axum/latest/axum/middleware/fn.from_fn.html), so it shouldn't be too difficult). You can wrap `predicate_required` too if you like.
